From 0dd131b02a3f2227fb062149e24c9e71b587a20b Mon Sep 17 00:00:00 2001
From: Marcelo Leitner <mleitner@redhat.com>
Date: Tue, 21 Oct 2014 16:21:24 -0400
Subject: [net] netfilter: nf_tables: nat expression must select CONFIG_NF_NAT

Message-id: <20141021162124.22888.46626.stgit@localhost.localdomain>
Patchwork-id: 98019
O-Subject: [RHEL7 net PATCH 122/124] netfilter: nf_tables: nat expression must select CONFIG_NF_NAT
Bugzilla: 1148041 1155088
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

Backport of upstream commit: 1e8430f30b55a1f3f6925c9f37f8cc9afd141d2e

 commit 1e8430f30b55a1f3f6925c9f37f8cc9afd141d2e
 Author: Pablo Neira Ayuso <pablo@netfilter.org>
 Date:   Mon Aug 11 18:21:50 2014 +0200

    netfilter: nf_tables: nat expression must select CONFIG_NF_NAT

    This enables the netfilter NAT engine in first place, otherwise
    you cannot ever select the nf_tables nat expression if iptables
    is not selected.

    Reported-by: Matteo Croce <technoboy85@gmail.com>
    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>

Signed-off-by: Marcelo Ricardo Leitner <mleitner@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/netfilter/Kconfig b/net/netfilter/Kconfig
index 6fc60a8..1abf802 100644
--- a/net/netfilter/Kconfig
+++ b/net/netfilter/Kconfig
@@ -508,7 +508,7 @@ config NFT_MASQ
 config NFT_NAT
  depends on NF_TABLES
  depends on NF_CONNTRACK
- depends on NF_NAT
+ select NF_NAT
  tristate "Netfilter nf_tables nat module"
  help
    This option adds the "nat" expression that you can use to perform
-- 
1.7.1